covid19/weather.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+"""Script to extract and transform weather data using the DarkSky API"""
+from typing import Optional
+
+import pandas as pd
+import requests
+
+from covid19 import utils
+from covid19.config import API_KEY
+
+
+def get_weather_data(api_token: str, lat: float, lon: float, date: str) -> dict:
+    """Returns historical weather conditions using the DarkSky API
+
+    Args:
+        api_token (str): DarkSky API key
+        lat (float): The latitude of a location
+        lon (float): The longitude of a location
+        date (str): Date of weather request
+
+    Returns:
+        dict: API response in JSON data format
+    """
+    # API requires date in ISO 8601 format
+    time = utils.get_iso_date(date)
+
+    url = f"https://api.darksky.net/forecast/{api_token}/{lat},{lon},{time}?exclude=currently,hourly,flags"
+    response = requests.get(url)
+    json_data = response.json()
+    print(response.status_code)
+    return json_data
+
+
+def extract_weather_data(weather_json: dict) -> Optional[pd.DataFrame]:
+    """Extracts data from the API response
+
+    Args:
+        weather_json (dict): [description]
+
+    Returns:
+        Optional[pd.DataFrame]: [description]
+    """
+    lat = weather_json.get('latitude', None)
+    lon = weather_json.get('longitude', None)
+    tz = weather_json.get('timezone', None)
+
+    # Data block containing weather conditions by day
+    daily_block = weather_json['daily'].get('data', [])
+
+    if len(daily_block) > 0:
```

Comment:
there's a way where you can make this into a general function so you're not adding an if statement for every field, but instead passing in a list of fields you expect to see and setting values based on that

Reply:
I like that idea. Can't picture that visually so would like to work on that.

Reply:
sounds good - add this to the todo for thursday